server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/DateHistogramAggregator.java

```diff
@@ -201,6 +226,79 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
         };
     }
 
+    private String fetchStarTreeCalendarUnit() {
+        CompositeDataCubeFieldType compositeMappedFieldType = (CompositeDataCubeFieldType) context.mapperService()
+            .getCompositeFieldTypes()
+            .iterator()
+            .next();
+        DateDimension starTreeDateDimension = (DateDimension) compositeMappedFieldType.getDimensions().get(0);
+
+        DateTimeUnitAdapter dateTimeUnitRounding = new DateTimeUnitAdapter(this.rounding.unit());
+        // check for null post this
+        DateTimeUnitRounding rounding = starTreeDateDimension.findClosestValidInterval(dateTimeUnitRounding);
+        String dimensionName = "@timestamp_" + rounding.shortName();
+        if (rounding.shortName().equals(this.rounding.unit().shortName())) {
+            this.starTreeDateRoundingRequired = false;
+        }
+        return dimensionName;
+    }
+
+    @Override
+    public StarTreeBucketCollector getStarTreeBucketCollector(LeafReaderContext ctx, CompositeIndexFieldInfo starTree) throws IOException {
+        return new StarTreeBucketCollector() {
+            {
+                this.starTreeValues = getStarTreeValues(ctx, starTree);
+                this.matchingDocsBitSet = StarTreeFilter.getPredicateValueToFixedBitSetMap(starTreeValues, starTreeDateDimension);
+                this.setSubCollectors();
+            }
+
+            public void setSubCollectors() throws IOException {
+                for (Aggregator aggregator : subAggregators) {
+                    this.subCollectors.add(((StarTreePreComputeCollector) aggregator).getStarTreeBucketCollector(ctx, starTree));
+                }
+            }
+
+            // TODO: fetch dimension name
+            SortedNumericStarTreeValuesIterator valuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+                .getDimensionValuesIterator(starTreeDateDimension);
+
+            String metricName = StarTreeUtils.fullyQualifiedFieldNameForStarTreeMetricsDocValues(
+                starTree.getField(),
+                "_doc_count",
+                MetricStat.DOC_COUNT.getTypeName()
+            );
+            SortedNumericStarTreeValuesIterator metricValuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
```

Comment:
If the metric values are guaranteed to be doc counts, this should be `docCountsIterator` and the `metricValue` variable should be called `docCount`.